### ISSUE TITLE
Hidden Properties should not validate in the form

### ISSUE THREAD

**Issue by kameelyan** [json-editor/json-editor#697]
### General information
- json-editor version: 2.1.0

### Expected behavior
Using options.hidden = true on a property alongside validation properties should cause the field to be ignored by validation.

### Actual behavior
Adding options.hidden = true to a string property alongside minLength = 1 causes the form to be invalid if the property is empty.

### Steps to reproduce the behavior

Direct link to example: https://json-editor.github.io/json-editor/?data=N4Ig9gDgLglmB2BnEAuUMDGCA2MBGqIAZglAIYDuApomALZUCsIANOHgFZUZQD62ZAJ5gArlELwwAJzplsrEIgwALKrNSgAJEtXqUIZVCgQUAelMda8ALQ61ZAHTSA5qYAmUskSjWADABZTO1kAYgUoQQgqQjBObnE2CClIKilYGg0QGDdM2ChsaP0ASQARcMjCxSgpGHhnBTpagBkqOqhlVABGNkhYBGQ0A2y3VtRqkSoAX0m2eDIGXJh8yoA5eei2CKjCRGra+rZG+Ba2jpROmZAyZ0LQPILCAEEb8u39WqgqG6kG5tbndpdaZsKRUACOIhgoJyKAA2lkcrN1gprtEALqXRDKMAUXipZJSAZZeCfTw8ODwcK6Sp4MBgKC7TwQfwgaZAA==
```
There are scenarios we're coming across where we want properties to have data that is generated on the back end (like ID). We don't want to manage separate schema validations for create/edit/view. ID is a property we want to hide from the user, to always exist in the data structure, and to have a minLength value of 1 when we validate the data after the system makes it's edits (generates an ID).

We can't achieve this currently because hidden fields in the form are causing the validation to fail. Given that hidden elements can't be unhidden by any means, I'd expect their validation to be skipped, or an option to allow for this.

**Comment by schmunk42** [comment]
I understand your use-case, in fact we had similar ones, but I don't think `hidden` should mean *do not validate*.

Is there something in the json-schema spec about it?

**Comment by kameelyan** [comment]
@schmunk42 I mean the issue is that options isn't part of a json-schema spec, right? That's a json-editor option that's added in?

**Comment by schmunk42** [comment]
Yeah right, there could be something like `skip_validation` in *options*.

@pmk65 @germanbisurgi Opinions?

**Comment by kameelyan** [comment]
I like that idea, could get behind that for sure.

**Comment by schmunk42** [comment]
That would be great ... PRs welcome, please include a test.

**Comment by germanbisurgi** [comment]
Hidden should not mean "not validate". "skip_validation" option is the way to go.
